### PR TITLE
r/aws_sns_topic_subscription: ignore `AuthorizationError` for `ListSubscriptionByTopic` operations

### LINKED ISSUE #42115: [Bug]: Breaking change with v5.94.0 and aws_sns_topic_subscription
### Terraform and AWS Provider Version

```console
1.9.7
5.94.0
```

### Affected Resource(s) or Data Source(s)

`aws_sns_topic_subscription`

### Expected Behavior

Cross account SNS subscriptions can refresh without error

### Actual Behavior

Refresh is not permitted by existing IAM policies

### Relevant Error/Panic Output

```console
AuthorizationError: User: arn:aws:sts::111111111111:assumed-role/abc123/terraform-run-V6rpqcU7sztpyrnS is not authorized to perform: SNS:ListSubscriptionsByTopic on resource: arn:aws:sns:us-west-2:222222222222:TopicName because no resource-based policy allows the SNS:ListSubscriptionsByTopic action
```

### Sample Terraform Configuration

<details>
<summary>Click to expand configuration</summary>

```hcl
resource "aws_sns_topic_subscription" "cross_account" {
  topic_arn = "arn:aws:sns:us-west-2:2222222222:TopicName "
  protocol  = "sqs"
  endpoint  = aws_sqs_queue.example.arn
}
```
</details>

### Steps to Reproduce

1. Create a topic in a secondary account that allows subscription from your primary account
2. Add subscription using Provider v5.93
3. Run plan
4. Upgrade Provider to 5.94
5. Run plan

### Debug Logging

<details>
<summary>Click to expand log output</summary>

```console

```
</details>

### GenAI / LLM Assisted Development

n/a

### Important Facts and References

_No response_

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by jar-b**
### Description

`v5.94.0` introduced an additional call to the `ListSubscriptionByTopic` API during the read operation to properly detect topic and subscription deletions which are only eventually consistent from the `GetSubscriptionAttributes` API. An unintended side effect of this addition was that existing principals which lack the appropriate IAM permissions to call `ListSubscriptionsByTopic` will observe apply time failures similar to:

```
AuthorizationError: User: arn:aws:sts::111111111111:assumed-role/abc123/terraform-run-V6rpqcU7sztpyrnS is not authorized to perform: SNS:ListSubscriptionsByTopic on resource: arn:aws:sns:us-west-2:222222222222:TopicName because no resource-based policy allows the SNS:ListSubscriptionsByTopic action
```

This change addresses the regression by skipping `AuthorizationError` exceptions rather than surfacing them as hard errors. The impact of ignoring these errors should be minimal, as the additional API call was only intended to catch eventual consistency edge cases that `GetSubscriptionAttributes` will still (eventually) catch.

### Relations

Closes #42115
Relates #42093 

### Output from Acceptance Testing

```console
% make testacc PKG=sns TESTS=TestAccSNSTopicSubscription_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.7 test ./internal/service/sns/... -v -count 1 -parallel 20 -run='TestAccSNSTopicSubscription_'  -timeout 360m -vet=off
2025/04/03 15:52:40 Initializing Terraform AWS Provider...

--- PASS: TestAccSNSTopicSubscription_filterPolicyScope_policyNotSet (3.73s)
--- PASS: TestAccSNSTopicSubscription_email (17.09s)
--- PASS: TestAccSNSTopicSubscription_autoConfirmingEndpoint (41.72s)
--- PASS: TestAccSNSTopicSubscription_disappears (50.27s)
--- PASS: TestAccSNSTopicSubscription_disappears_TopicExternal (50.46s)
--- PASS: TestAccSNSTopicSubscription_disappears_Topic (50.53s)
--- PASS: TestAccSNSTopicSubscription_basic (52.11s)
--- PASS: TestAccSNSTopicSubscription_rawMessageDelivery (70.27s)
--- PASS: TestAccSNSTopicSubscription_deliveryPolicy (70.86s)
--- PASS: TestAccSNSTopicSubscription_filterPolicy (71.45s)
--- PASS: TestAccSNSTopicSubscription_autoConfirmingSecuredEndpoint (85.36s)
--- PASS: TestAccSNSTopicSubscription_firehose (95.70s)
--- PASS: TestAccSNSTopicSubscription_redrivePolicy (104.99s)
--- PASS: TestAccSNSTopicSubscription_filterPolicyScope (148.66s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/sns        155.739s
```

### CODE REVIEW

File: .changelog/42117.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/aws_service_thing: Ignore `AuthorizationError` exceptions for `ListSubscriptionByTopic` operations. This fixes a regression introduced in [`v5.94.0`](https://github.com/hashicorp/terraform-provider-aws/pull/42093).
```
Comment (ewbankkit) [inline .changelog/42117.txt:2]:
```suggestion
resource/aws_sns_topic_subscription: Ignore `AuthorizationError` exceptions for `ListSubscriptionByTopic` operations. This fixes a regression introduced in [`v5.94.0`](https://github.com/hashicorp/terraform-provider-aws/pull/42093).
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

**Review by nam054 — APPROVED** [review]
LGTM
